include/swift/SIL/AbstractionPattern.h

```diff
@@ -354,17 +383,15 @@ class AbstractionPattern {
   }
 
   bool hasGenericSignature() const {
-    return (getKind() == Kind::Type ||
-            getKind() == Kind::Discard ||
-            hasStoredClangType() ||
-            hasStoredObjCMethod());
+    return (getKind() == Kind::Type || getKind() == Kind::Discard ||
+            hasStoredClangType() || hasStoredObjCMethod() ||
+            hasStoredCXXMethod());
```

Comment:
Could you just turn this into an exhaustive switch?  I don't think the succession of helper method calls is doing much to make this more maintainable.